### PR TITLE
Fix TestConnectivityOnStartup

### PR DESCRIPTION
**Pull request by at-wat**
It failed when accepting routine is not started faster than dial.

### Reference issue
Part of #204

### CODE REVIEW

File: connectivity_vnet_test.go [hunk 1]
```diff
@@ -169,15 +169,33 @@ func connectWithVNet(aAgent, bAgent *Agent) (*Conn, *Conn) {
...
+	accepting := make(chan struct{})
 	var aConn *Conn
 
+	origHdlr := aAgent.onConnectionStateChangeHdlr.Load()
+	if origHdlr != nil {
+		defer func() {
+			_ = aAgent.OnConnectionStateChange(origHdlr.(func(ConnectionState)))
+		}()
+	}
+	_ = aAgent.OnConnectionStateChange(func(s ConnectionState) {
```
Comment (Sean-Der) [inline connectivity_vnet_test.go:181]: Mind wrapping everything in a `check` instead of `_ =`?

**Comment by Sean-Der** [comment]
@at-wat oh actually my mistake, I don't think we should use `connectWithVNet` at all!

I would duplicate this all just for this one instance. Adding this assertion to `connectWithVNet` might cover up other issues. This single test is just to make sure we send messages even without a `taskLoopInterval`

**Comment by at-wat** [comment]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x14 pc=0x831bc82]

goroutine 839 [running]:
github.com/pion/ice/v2.(*Agent).handleInbound(0xa88e900, 0xa854630, 0x8497da0, 0xa85aaa0, 0x8492810, 0xa9cfd00)
	/go/src/github.com/pion/ice/agent.go:924 +0x672
github.com/pion/ice/v2.handleInboundCandidateMsg.func1(0xa88e900)
	/go/src/github.com/pion/ice/candidate_base.go:113 +0x44
github.com/pion/ice/v2.(*Agent).run(0xa88e900, 0xab80778, 0xaa6c600, 0xa9e6310, 0x0)
	/go/src/github.com/pion/ice/agent.go:169 +0x1d2
github.com/pion/ice/v2.handleInboundCandidateMsg(0x8497da0, 0xa85aaa0, 0xaa86000, 0x70, 0x2000, 0x8492810, 0xa9cfd00, 0x8496ca0, 0xa9cf020)
	/go/src/github.com/pion/ice/candidate_base.go:112 +0x28f
github.com/pion/ice/v2.(*candidateBase).recvLoop(0xa85aaa0)
	/go/src/github.com/pion/ice/candidate_base.go:97 +0xc6
created by github.com/pion/ice/v2.(*candidateBase).start
	/go/src/github.com/pion/ice/candidate_base.go:81 +0xc0
FAIL	github.com/pion/ice/v2	64.697s
FAIL
```
#202 happened on CI.

**Comment by at-wat** [comment]
@Sean-Der reverted `connectivity_vnet_test.go` and move fixed code inside `TestConnectivityOnStartup`.

**Pull request merged**